topic_dynamics/modeling.py

```diff
@@ -0,0 +1,273 @@
+    plt.xlabel("Iterations count")
+    plt.ylabel("Phi Sparsity")
+    plt.grid(True)
+    plt.savefig(os.path.abspath(os.path.join(directory, "results", name + "_phi_sparsity.png")), dpi = 1200)
+    plt.close()
+
+    plt.plot(range(model.num_phi_updates),
+             model.score_tracker["SparsityThetaScore"].value, "r--", linewidth=2)
+    plt.xlabel("Iterations count")
+    plt.ylabel("Theta Sparsity")
+    plt.grid(True)
+    plt.savefig(os.path.abspath(os.path.join(directory, "results", name + "_theta_sparsity.png")), dpi = 1200)
+    plt.close()
+
+
+def save_most_popular_tokens(model: artm.artm_model.ARTM, directory: str, name: str) -> None:
+    """
+    Save the most popular tokens of the model.
+    :param model: the model.
+    :param directory: the directory with the dataset.
+    :param name: name of the processed dataset.
+    :return: None.
+    """
+    with open(os.path.abspath(os.path.join(directory, "results", name + "_most_popular_tokens.txt")), "w+") as fout:
+        for topic_name in model.topic_names:
+            fout.write(topic_name + " : " + str(model.score_tracker["TopTokensScore"].last_tokens[topic_name]) + "\n")
+
+
+def save_matrices(model: artm.artm_model.ARTM, directory: str, name: str) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """
+    Save the Phi and Theta matrices.
+    :param model: the model.
+    :param directory: the directory with the dataset.
+    :param name: name of the processed dataset.
+    :return: Two matrices as DataFrames.
+    """
+    phi_matrix = model.get_phi().sort_index(axis=0)
+    phi_matrix.to_csv(os.path.abspath(os.path.join(directory, "results", name + "_phi.csv")))
+    theta_matrix = model.get_theta().sort_index(axis=1)
+    theta_matrix.to_csv(os.path.abspath(os.path.join(directory, "results", name + "_theta.csv")))
+    return phi_matrix, theta_matrix
+
+
+def save_most_topical_files(number_of_topics: int, theta: pd.DataFrame, directory: str, name: str) -> None:
+    """
+    Save the most topical files of the model.
+    :param number_of_topics: the number of topics.
+    :param theta: Theta matrix.
+    :param directory: the directory with the dataset.
+    :param name: name of the processed dataset.
+    :return: None.
+    """
+    file_address = {}
+    with open(os.path.abspath(os.path.join(directory, name + "_tokens.txt")), "r") as fin:
+        for line in fin:
+            file_address[int(line.split(";")[0])] = line.split(";")[1]
+    with open(os.path.abspath(os.path.join(directory, "results", name + "_most_topical_files.txt")), "w+") as fout:
+        for i in range(1, number_of_topics + 1):
+            fout.write("Topic " + str(i) + "\n\n")
+            dictionary_of_the_topic = theta.sort_values(by="topic_" + str(i), axis=1,
+                                                        ascending=False).loc["topic_" + str(i)][:10].to_dict()
+            for j in dictionary_of_the_topic.keys():
```

Comment:
```suggestion
            for k in dictionary_of_the_topic.keys():
```
because it's iteration over keys - it will be a bit nicer